OpenRA.Game/Network/LocalizedMessage.cs

```diff
@@ -0,0 +1,119 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Fluent.Net;
+using OpenRA.Widgets;
+
+namespace OpenRA.Network
+{
+	public class FluentArgument
+	{
+		[Flags]
+		public enum FluentArgumentType
+		{
+			String = 0,
+			Number = 1,
+		}
+
+		public readonly string Key;
+		public readonly string Value;
+		public readonly FluentArgumentType Type;
+
+		public FluentArgument() { }
+
+		public FluentArgument(string key, object value)
+		{
+			Key = key;
+			Value = value.ToString();
+			Type = value is sbyte || value is short || value is int || value is long ||
+				value is byte || value is ushort || value is uint || value is ulong ||
+				value is float || value is double || value is decimal ?
+				FluentArgumentType.Number : FluentArgumentType.String;
```

Comment:
Something like this could be cleaner,

```cs
		FluentArgumentType GetFluentArgumentType(object value)
		{
			switch (value)
			{
				case sbyte b:
				case short s:
					return FluentArgumentType.Number;
				default:
					return FluentArgumentType.String;
			}
		}
```

Reply:
I came up with something that does not assign the unused variable.

Reply:
I dont know if you saw that I updated the example with using discard, and if we one day update to c# 9 we can just remove the discard for even cleaner code.

Reply:
Updated using the `_`